sdk/core/core/inc/az_http_transport.h

```diff
@@ -85,16 +85,19 @@ AZ_NODISCARD az_result
 az_http_request_get_header(_az_http_request const* request, int32_t index, az_pair* out_header);
 
 /**
- * @brief Get parts of an HTTP request. NULL in accepted to ignore getting any parts, for example,
- * call this function like below to get only the http method and ignore getting url and body
- *   az_http_request_get_parts(request, &method, NULL, NULL)`
+ * @brief Get parts of an HTTP request. `NULL` in accepted to ignore getting any parts, for example,
+ * call this function like below to get only the http method and ignore getting url and body.
+ *   `az_http_request_get_parts(request, &method, `NULL`, `NULL`)`
  *
  * This function is expected to be used by transport layer only.
  *
  * @param request HTTP request to get parts from.
- * @param out_method Pointer to write HTTP method to. Use NULL to ignore getting this value.
- * @param out_url Pointer to write URL to. Use NULL to ignore getting this value.
- * @param out_body Pointer to write HTTP request body to. Use NULL to ignore getting this value.
+ * @param out_method[out] __[nullable]__ Pointer to write HTTP method to. Use `NULL` to ignore
```

Comment:
The direction [in]/[out] goes right after `@param`.

It should follow this format (from the design guidelines):
`@param[<direction>] <param_name> __<additional_annotations>__ description`

Reply:
please feel free to make any change on documentation directly into my branches XD lol.